### PR TITLE
EWPP-288: Add contacts field to news content type

### PR DESCRIPTION
**Pull request by enriquelacoma**
### Description

Add field oe_news_contacts to news content type.

### Change log

- Added: oe_news_contacts field and hook_post_udpates_N().

### Commands

```sh
If the module oe_content_news is already installed, run drush up

```

### CODE REVIEW

File: modules/oe_content_news/config/install/field.field.node.oe_news.field_oe_news_contacts.yml [hunk 1]
```diff
@@ -0,0 +1,35 @@
+langcode: en
```
Comment (yenyasinn) [inline modules/oe_content_news/config/install/field.field.node.oe_news.field_oe_news_contacts.yml:1]: Has to be renamed to oe_news_contacts

File: tests/features/news.feature [hunk 2]
```diff
@@ -4,9 +4,12 @@ Feature: News content creation
   As an editor
   I need to be able to create and see news items
 
-  @cleanup:media
+  @javascript @cleanup:media
```
Comment (yenyasinn) [inline tests/features/news.feature:7]: `@javascript` should be removed

File: modules/oe_content_news/config/install/field.field.node.oe_news.field_oe_news_contacts.yml [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+description: ''
+required: false
+translatable: false
+default_value: {  }
+default_value_callback: ''
+settings:
+  handler: 'default:oe_contact'
+  handler_settings:
+    target_bundles:
+      oe_general: oe_general
```
Comment (yenyasinn) [inline modules/oe_content_news/config/install/field.field.node.oe_news.field_oe_news_contacts.yml:29]: Only General bundle is needed.

Comment (enriquelacoma) [reply modules/oe_content_news/config/install/field.field.node.oe_news.field_oe_news_contacts.yml]: Updated

File: modules/oe_content_news/config/install/field.field.node.oe_news.field_oe_news_contacts.yml (hunk 1 above)
Comment (enriquelacoma) [reply modules/oe_content_news/config/install/field.field.node.oe_news.field_oe_news_contacts.yml]: Updated

File: tests/features/news.feature (hunk 2 above)
Comment (enriquelacoma) [reply tests/features/news.feature:7]: The Contacts field uses javascript, we need the tag to use it on the test.

File: modules/oe_content_news/config/post_updates/00007_update_view_display/core.entity_view_display.node.oe_news.default.yml [hunk 4]
```diff
@@ -0,0 +1,76 @@
+langcode: en
```
Comment (yenyasinn) [inline modules/oe_content_news/config/post_updates/00007_update_view_display/core.entity_view_display.node.oe_news.default.yml:1]: we don't update default view mode in oe_content. It has to be done in oe_theme.

Comment (enriquelacoma) [reply modules/oe_content_news/config/post_updates/00007_update_view_display/core.entity_view_display.node.oe_news.default.yml:1]: To show the Contacts field on news, without using oe_theme, we need to update the default view for the content type. Should we not show Contacts field when not using oe_theme?

Comment (yenyasinn) [reply modules/oe_content_news/config/post_updates/00007_update_view_display/core.entity_view_display.node.oe_news.default.yml:1]: If oe_theme isn't used it means that custom default view mode is used on the site. We shouldn't override it.

Comment (enriquelacoma) [reply modules/oe_content_news/config/post_updates/00007_update_view_display/core.entity_view_display.node.oe_news.default.yml]: Update removed

File: modules/oe_content_news/oe_content_news.post_update.php [hunk 5]
```diff
@@ -57,3 +57,28 @@ function oe_content_news_post_update_00004(): void {
     $updated_form_display->save();
   }
 }
+
+/**
+ * Create oe_news_contacts field in the news content type.
+ */
+function oe_content_news_post_update_00005(): void {
+  $storage = new FileStorage(drupal_get_path('module', 'oe_content_news') . '/config/post_updates/00005_create_fields');
```
Comment (yenyasinn) [inline modules/oe_content_news/oe_content_news.post_update.php:65]: We can move these new fields to oe_content_news_post_update_00003().

File: modules/oe_content_news/oe_content_news.post_update.php [hunk 6]
```diff
@@ -57,3 +57,28 @@ function oe_content_news_post_update_00004(): void {
...
+}
+
+/**
+ * Update news node form display.
+ */
+function oe_content_news_post_update_00006(): void {
+  $storage = new FileStorage(drupal_get_path('module', 'oe_content_news') . '/config/post_updates/00006_update_form_display');
+
+  // Form display configurations to update.
+  $form_display_values = $storage->read('core.entity_form_display.node.oe_news.default');
```
Comment (yenyasinn) [inline modules/oe_content_news/oe_content_news.post_update.php:76]: Please update in core.entity_form_display.node.oe_news.default 00004_update_form_display and we can get rid of oe_content_news_post_update_00006()

File: modules/oe_content_news/oe_content_news.info.yml [hunk 7]
```diff
@@ -10,6 +10,9 @@ dependencies:
   - oe_content:oe_content_reference_code_field
   - oe_media:oe_media
   - oe_media_avportal:oe_media_avportal
+  - oe_content_entity_contact:oe_content_entity_contact
```
Comment (yenyasinn) [inline modules/oe_content_news/oe_content_news.info.yml:13]: These modules have to be added to oe_content_news_post_update_00002()

File: modules/oe_content_news/oe_content_news.post_update.php (hunk 5 above)
Comment (enriquelacoma) [reply modules/oe_content_news/oe_content_news.post_update.php]: updated

File: modules/oe_content_news/oe_content_news.post_update.php (hunk 6 above)
Comment (enriquelacoma) [reply modules/oe_content_news/oe_content_news.post_update.php]: updated

File: modules/oe_content_news/oe_content_news.info.yml (hunk 7 above)
Comment (enriquelacoma) [reply modules/oe_content_news/oe_content_news.info.yml:13]: updated

**Review by 22Alexandra — DISMISSED** [review]

File: modules/oe_content_news/config/install/field.storage.node.oe_news_contacts.yml [hunk 8]
```diff
@@ -0,0 +1,20 @@
...
+id: node.oe_news_contacts
+field_name: oe_news_contacts
+entity_type: node
+type: entity_reference_revisions
+settings:
+  target_type: oe_contact
+module: entity_reference_revisions
+locked: false
+cardinality: -1
+translatable: true
```
Comment (ademarco) [inline modules/oe_content_news/config/install/field.storage.node.oe_news_contacts.yml:17]:
```suggestion
translatable: false
```

File: modules/oe_content_news/config/post_updates/00003_create_fields/field.storage.node.oe_news_contacts.yml [hunk 9]
```diff
@@ -0,0 +1,20 @@
...
+id: node.oe_news_contacts
+field_name: oe_news_contacts
+entity_type: node
+type: entity_reference_revisions
+settings:
+  target_type: oe_contact
+module: entity_reference_revisions
+locked: false
+cardinality: -1
+translatable: true
```
Comment (ademarco) [inline modules/oe_content_news/config/post_updates/00003_create_fields/field.storage.node.oe_news_contacts.yml:17]:
```suggestion
translatable: false
```

File: modules/oe_content_news/oe_content_news.info.yml [hunk 10]
```diff
@@ -10,6 +10,9 @@ dependencies:
...
+  - composite_reference:composite_reference
+  - entity_reference_revisions:entity_reference_revisions
```
Comment (ademarco) [inline modules/oe_content_news/oe_content_news.info.yml:15]: `inline_entity_form` is also required now. Also update readme for the submodule, including possible patches.

**Review by ademarco — CHANGES_REQUESTED** [review]

File: modules/oe_content_news/config/install/field.storage.node.oe_news_contacts.yml (hunk 8 above)
Comment (enriquelacoma) [reply modules/oe_content_news/config/install/field.storage.node.oe_news_contacts.yml]: Updated

File: modules/oe_content_news/config/post_updates/00003_create_fields/field.storage.node.oe_news_contacts.yml (hunk 9 above)
Comment (enriquelacoma) [reply modules/oe_content_news/config/post_updates/00003_create_fields/field.storage.node.oe_news_contacts.yml]: Updated

File: modules/oe_content_news/oe_content_news.info.yml (hunk 10 above)
Comment (enriquelacoma) [reply modules/oe_content_news/oe_content_news.info.yml:15]: Updated and readme added

File: modules/oe_content_news/oe_content_news.post_update.php [hunk 11]
```diff
@@ -30,11 +30,20 @@ function oe_content_news_post_update_00001_update_field_labels(array &$sandbox):
  * Enable new modules from dependency.
  */
 function oe_content_news_post_update_00002(): void {
-  \Drupal::service('module_installer')->install(['field_group', 'oe_content_reference_code_field']);
+  $modules = [
+    'field_group',
+    'oe_content_reference_code_field',
+    'oe_content_entity_contact',
+    'composite_reference',
+    'entity_reference_revisions',
```
[the remark this answers is not in the document]
Comment (enriquelacoma) [reply modules/oe_content_news/oe_content_news.post_update.php:38]: Updated

**Review by 22Alexandra — DISMISSED** [review]

**Review by ademarco — APPROVED** [review]

**Review by 22Alexandra — APPROVED** [review]

**Pull request merged**